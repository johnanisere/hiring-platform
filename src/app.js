'use strict';
exports.__esModule = true;
var http_errors_1 = require('http-errors');
var express_1 = require('express');
var cors_1 = require('cors');
var compression_1 = require('compression');
var morgan_1 = require('morgan');
var express_graphql_1 = require('express-graphql');
var path_1 = require('path');
var mongoose_1 = require('mongoose');
var index_1 = require('./routes/index');
var schema_1 = require('./schema');
var app = express_1['default']();
// Setup Request logging
var logFormat = process.env.NODE_ENV === 'production' ? 'combined' : 'dev';
app.use(
  morgan_1['default'](logFormat, {
    skip: function(_req, res) {
      if (process.env.NODE_ENV === 'test') {
        return true;
      }
      return res.statusCode < 400;
    },
    stream: process.stderr,
  }),
);
app.use(
  morgan_1['default'](logFormat, {
    skip: function(_req, res) {
      if (process.env.NODE_ENV === 'test') {
        return true;
      }
      return res.statusCode >= 400;
    },
    stream: process.stdout,
  }),
);
app.disable('x-powered-by');
app.use(compression_1['default']());
app.use(cors_1['default']());
app.use(express_1['default'].json());
app.use(express_1['default'].urlencoded({ extended: false }));
app.use('/api', index_1['default']);
mongoose_1['default'].connect(
  'mongodb://localhost:27017/test',
  { useNewUrlParser: true },
  function() {
    console.log('Database connected succesfully');
  },
);
var connection = mongoose_1['default'].connection;
connection.once('open', function() {
  console.log('MongoDB database connection established successfully');
});
app.use(
  '/graphql',
  express_graphql_1['default']({
    schema: schema_1['default'],
    graphiql: true,
  }),
);
if (process.env.NODE_ENV === 'production') {
  app.use(
    express_1['default'].static(
      path_1['default'].join(__dirname, '../', 'client/build'),
    ),
  );
  app.get('/*', function(_req, res) {
    res.sendFile(
      path_1['default'].join(__dirname, '../', 'client/build/index.html'),
    );
  });
}
// catch 404 and forward to error handler
app.use(function(_req, _res, next) {
  next(http_errors_1['default'](404));
});
// error handler
app.use(function(err, req, res) {
  // set locals, only providing error in development
  res.locals.message = err.message;
  res.locals.error = req.app.get('env') === 'development' ? err : {};
  // render the error page
  res.status(err.status || 500);
  res.render('error');
});
exports['default'] = app;
