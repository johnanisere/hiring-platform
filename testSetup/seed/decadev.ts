import User from '../../src/models/User';

const users = [
  {
    email: 'doe@example.com',
    name: 'John Doe',
    phone: '08074583218',
    password: 'mysecret2',
    role: 'admin',
    profilePhoto:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSIcqjP8a8EfGrtkCX0CJXJ4HRltM5u8Jdhkz_TneWdb4rfPsMi',
  },

  {
    email: 'decadev@example.com',
    name: 'Jane Mary',
    phone: '08074583218',
    password: 'mysecret2',
    role: 'dev',
    profilePhoto:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSIcqjP8a8EfGrtkCX0CJXJ4HRltM5u8Jdhkz_TneWdb4rfPsMi',
    pod: 'android',
  },
];

export const seedUsers = async () => {
  try {
    const allUsers = users.map(async user => {
      const newUser = new User(user);
      return newUser.save();
    });

    const res = await Promise.all(allUsers);
    return res;
  } catch (err) {
    return err;
  }
};

const seed = async () => {
  try {
    const res = await seedUsers();
    console.log(`Database has been seeded with ${res.length} users`);
  } catch (err) {
    console.log({ err });
  }
};

export default seed;
