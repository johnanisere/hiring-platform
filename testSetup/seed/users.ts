import User from '../../src/models/User';
import HiringPartner from '../../src/models/HiringPartner';

const users = [
  {
    email: 'doe@example.com',
    name: 'John Doe',
    phone: '08074583218',
    password: 'mysecret2',
    role: 'admin',
    profilePhoto:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSIcqjP8a8EfGrtkCX0CJXJ4HRltM5u8Jdhkz_TneWdb4rfPsMi',
  },

  {
    email: 'decadev-user@example.com',
    name: ' Test Decadev',
    phone: '08074583218',
    password: 'mysecret',
    role: 'dev',
    profilePhoto:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSIcqjP8a8EfGrtkCX0CJXJ4HRltM5u8Jdhkz_TneWdb4rfPsMi',
    pod: 'java',
  },
  {
    email: 'decadev-use2r@example.com',
    name: ' Test Decadev',
    phone: '08074583208',
    password: 'mysecret',
    role: 'dev',
    profilePhoto:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSIcqjP8a8EfGrtkCX0CJXJ4HRltM5u8Jdhkz_TneWdb4rfPsMi',
    pod: 'nodejs',
  },
];

const hirers = [
  {
    name: 'Dola Akingbola',
    email: 'dola@example.com',
    nameOfOrg: 'Twitter',
    designation: 'CTO',
    website: 'https://www.google.com',
    industry: 'Technology',
    phone: '08066589871',
    numberOfTalentsRequired: '6-10',
    deadline: 'Within 1 Month',
    verified: true,
    active: true,
    password: 'mysecret',
    interestLanguage: [],
    interviews: [],
  },
  {
    name: 'Test Hirer',
    email: 'hirer2@example.com',
    nameOfOrg: 'Paystack',
    designation: 'CTO',
    website: 'https://www.google.com',
    industry: 'Technology',
    phone: '08066559871',
    numberOfTalentsRequired: '6-10',
    deadline: 'Within 1 Month',
    verified: false,
    active: false,
    password: 'mysecret',
    interestLanguage: [],
    interviews: [],
  },
];

export const seedUsers = async () => {
  try {
    const allUsers = users.map(async user => {
      const newUser = new User(user);
      return newUser.save();
    });

    const res = await Promise.all(allUsers);
    return res;
  } catch (err) {
    return err;
  }
};

export const seedPartners = async () => {
  try {
    const allPartners = hirers.map(async partner => {
      const newPartner = new HiringPartner(partner);
      return await newPartner.save();
    });
    const res = await Promise.all(allPartners);
    return res;
  } catch (err) {
    return err;
  }
};

const seed = async () => {
  try {
    const res = await seedUsers();
    console.log(`Database has been seeded with ${res.length} users`);
    const partners = await seedPartners();
    console.log(`Database has been seeded with ${partners.length} partner(s)`);
  } catch (err) {
    console.log({ err });
  }
};

export default seed;
