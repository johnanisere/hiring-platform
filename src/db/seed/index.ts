import users from './data/users';
import User from '../../models/User';
import Cycle from '../../models/Cycle';
import Employment from '../../models/Employment';
import employments from './data/employments';

const cleanDb = async () => {
  try {
    await Cycle.deleteMany({});
    await User.deleteMany({});
    return console.log('succesfully cleared db');
  } catch (err) {
    console.log('Error: occured', err);
    return err;
  }
};

export const seedUsers = async () => {
  try {
    const allUsers = users.map(async user => {
      const newUser = await new User(user);
      return newUser.save();
    });
    const cycle = await new Cycle();
    await cycle.save();
    const res = await Promise.all(allUsers);
    return res;
  } catch (err) {
    return err;
  }
};

export const seedEmployments = async () => {
  try {
    const allEmployments = employments.map(async employment => {
      const newEmployment = await new Employment(employment);
      return newEmployment.save();
    });
    const res: any = await Promise.all(allEmployments);
    const dev: any = await User.findOne({ email: 'sheyiogundijo@gmail.com' });
    res.forEach((employment: any) => dev.employments.push(employment._id));
    await dev.save();
    return res;
  } catch (err) {
    return err;
  }
};

const seed = () => {
  cleanDb()
    .then(() => {
      return seedUsers();
    })
    .then(() => {
      return seedEmployments();
    })
    .then(res => {
      console.log(`Database has been seeded with ${res.length} users`);
    })
    .catch(err => {
      console.log({ err });
    });
};

export default seed;
