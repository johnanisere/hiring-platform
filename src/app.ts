import dotenv from 'dotenv';
import createError from 'http-errors';
import express from 'express';
import cors from 'cors';
import compression from 'compression';
import morgan from 'morgan';
import graphQLHTTP from 'express-graphql';
import path from 'path';
import seed from './db/seed';
import dbConnection from './db';
import apiRouter from './routes/index';
import usersRouter from './routes/users';
import schema from './schema';

dotenv.config();
const app = express();

dbConnection.once('open', function() {
  seed();
  console.log('MongoDB database connection established successfully');
});

// Setup Request logging
const logFormat = process.env.NODE_ENV === 'production' ? 'combined' : 'dev';

app.use(
  morgan(logFormat, {
    skip: function(_req, res) {
      if (process.env.NODE_ENV === 'test') {
        return true;
      }

      return res.statusCode < 400;
    },
    stream: process.stderr,
  }),
);

app.use(
  morgan(logFormat, {
    skip: function(_req, res) {
      if (process.env.NODE_ENV === 'test') {
        return true;
      }

      return res.statusCode >= 400;
    },
    stream: process.stdout,
  }),
);

app.disable('x-powered-by');
app.use(compression());
app.use(cors());
app.use(express.json());
app.use(express.urlencoded({ extended: false }));

app.use('/api', apiRouter);
app.use('/users', usersRouter);

app.use(
  '/graphql',
  graphQLHTTP({
    schema,
    graphiql: true,
  }),
);

if (process.env.NODE_ENV === 'production') {
  app.use(express.static(path.join(__dirname, '../', 'client/build')));
  app.get('/*', (_req, res) => {
    res.sendFile(path.join(__dirname, '../', 'client/build/index.html'));
  });
}

// catch 404 and forward to error handler
app.use(function(
  _req: express.Request,
  _res: express.Response,
  next: express.NextFunction,
) {
  next(createError(404));
});

// error handler
app.use(function(err: any, req: express.Request, res: express.Response) {
  // set locals, only providing error in development
  res.locals.message = err.message;
  res.locals.error = req.app.get('env') === 'development' ? err : {};

  // render the error page
  res.status(err.status || 500);
  res.render('error');
});

export default app;
