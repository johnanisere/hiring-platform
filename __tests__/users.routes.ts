import request from 'supertest';

import app from '../src/app';
import seed from '../testSetup/seed/users';

const { connectMongoDB, disconnectMongoDB } = require('../testSetup/mongodb');

const credentials = {
  email: 'doe@example.com',
  password: 'mysecret2',
};
const hirerCredentials = {
  email: 'dola@example.com',
  password: 'mysecret',
};
const credentials2 = {
  email: 'doe@example.com',
  password: 'mysecret222',
};
const decadevCredentials = {
  email: 'decadev-user@example.com',
  password: 'mysecret',
};

let hirer: any;
let hirerToken: string;
let decadev: any;
let decadevToken: string;
let employmentId: string;
let skillId: string;
let projectId: string;

beforeAll(async () => {
  await connectMongoDB('users-test');
  await seed();

  hirer = await request(app)
    .post('/api/v1/hirer/login')
    .send(hirerCredentials);

  hirerToken = hirer.body.token;

  decadev = await request(app)
    .post('/api/v1/users/login')
    .send(decadevCredentials);

  decadevToken = decadev.body.token;
});

afterAll(async () => await disconnectMongoDB());

describe('/users/', () => {
  test('/users/login', async () => {
    expect.assertions(4);
    return request(app)
      .post('/api/v1/users/login')
      .send(credentials)
      .expect(res => {
        expect(res.status).toBe(200);
        expect(Object.keys(res.body)).toContain('token');
        expect(Object.keys(res.body)).toContain('email');
        expect(Object.keys(res.body)).toContain('name');
      });
  });
  test('/users/login ==> Returns wrong password if password is wrong', async () => {
    expect.assertions(1);
    return request(app)
      .post('/api/v1/users/login')
      .send(credentials2)
      .expect(res => {
        expect(res.body.error).toBe('wrong password');
      });
  });

  test('lists java decadevs', async () => {
    expect.assertions(1);
    const pod = 'java';
    const decadevs = await request(app)
      .get(`/api/v1/users/decadevs?pod=${pod}`)
      .set('Authorization', `Bearer ${hirerToken}`);

    for (let i = 0; i < JSON.parse(decadevs.text).allDecadevs.length; i++) {
      expect(JSON.parse(decadevs.text).allDecadevs[i]).toHaveProperty(
        'pod',
        'java',
      );
    }
  });

  test('lists nodejs decadevs', async () => {
    expect.assertions(1);
    const pod = 'nodejs';
    const decadevs = await request(app)
      .get(`/api/v1/users/decadevs?pod=${pod}`)
      .set('Authorization', `Bearer ${hirerToken}`);

    for (let i = 0; i < JSON.parse(decadevs.text).allDecadevs.length; i++) {
      expect(JSON.parse(decadevs.text).allDecadevs[i]).toHaveProperty(
        'pod',
        'nodejs',
      );
    }
  });
});

describe('Updates Decadev Profile', () => {
  test('Updates profile/contact details of Decadevs', async () => {
    expect.assertions(7);
    return request(app)
      .put(`/api/v1/users/update/userInfo/${decadevCredentials.email}`)
      .set('Authorization', `Bearer ${decadevToken}`)
      .send({
        name: 'Jane Mary Lala',
        location: 'Lagos',
        profilePhoto:
          'https://encq=tbn:ANd9GcSIcqjP8a8EfGrtkCX0CJXJ4HRltM5u8Jdhkz_TneWdb4rfPsMi',
        bio: 'A very good dev',
        github: 'https://github.com',
        linkedIn: 'https://linkedin.com/in/oluwaseyi-ogundijo-a29699195',
        stackOverflow: 'https://stackoverflow.com',
        website: 'https://www.google.com/',
      })
      .expect(res => {
        expect(res.body.message).toEqual(expect.any(String));
        expect(Object.keys(res.body)).toContain('message');
        expect(Object.keys(res.body)).toContain('user');

        expect(Object.keys(res.body.user)).toContain('bio');
        expect(Object.keys(res.body.user)).toContain('github');
        expect(Object.keys(res.body.user)).toContain('email');
        expect(Object.keys(res.body.user)).toContain('phone');
      });
  });

  test('Adds new employment details of Decadevs', async () => {
    expect.assertions(7);
    return request(app)
      .put(`/api/v1/users/update/new-employment/${decadevCredentials.email}`)
      .set('Authorization', `Bearer ${decadevToken}`)
      .send({
        achievements: [
          'Built a language-learning app for Arabic learners in AngularJS',
          "Constructed a reporting tool so administrators could see their students' progress in real-time.",
          'Developed custom AngularJS services and corresponding tests.',
          'Created custom exact-target email templates and data extensions, and a tool to allow administrators to email users at the click of a button.',
        ],
        title: 'CTO',
        location: 'Googlee',
        duration: '2018 (9 months)',
      })
      .expect(res => {
        employmentId = res.body.user.employments[0]._id;
        expect(res.body.message).toEqual(expect.any(String));
        expect(Object.keys(res.body)).toContain('message');
        expect(Object.keys(res.body)).toContain('user');

        expect(Object.keys(res.body.user)).toContain('bio');
        expect(Object.keys(res.body.user)).toContain('github');
        expect(Object.keys(res.body.user)).toContain('email');
        expect(Object.keys(res.body.user)).toContain('phone');
      });
  });

  test('Adds new skill details of Decadevs', async () => {
    expect.assertions(8);
    return request(app)
      .put(`/api/v1/users/update/new-skill/${decadevCredentials.email}`)
      .set('Authorization', `Bearer ${decadevToken}`)
      .send({
        type: 'Food',
        description: 'Cakes, Chocolate, Desert',
      })
      .expect(res => {
        skillId = res.body.user.skills[0]._id;
        const { skills, employments, ...rest } = res.body.user;
        expect(res.body.message).toEqual(expect.any(String));
        expect(Object.keys(res.body)).toContain('message');
        expect(Object.keys(res.body)).toContain('user');

        expect(Object.keys(res.body.user)).toContain('bio');
        expect(Object.keys(res.body.user)).toContain('github');
        expect(Object.keys(res.body.user)).toContain('email');
        expect(Object.keys(res.body.user)).toContain('phone');
        expect(rest).toMatchSnapshot();
      });
  });

  test('Adds new project details of Decadevs', async () => {
    expect.assertions(7);
    return request(app)
      .put(`/api/v1/users/update/new-project/${decadevCredentials.email}`)
      .set('Authorization', `Bearer ${decadevToken}`)
      .send({
        title: 'Twitter',
        languages: 'Visual Studio, HTML, Sitecore',
      })
      .expect(res => {
        projectId = res.body.user.portfolio[0]._id;
        expect(res.body.message).toEqual(expect.any(String));
        expect(Object.keys(res.body)).toContain('message');
        expect(Object.keys(res.body)).toContain('user');

        expect(Object.keys(res.body.user)).toContain('bio');
        expect(Object.keys(res.body.user)).toContain('github');
        expect(Object.keys(res.body.user)).toContain('email');
        expect(Object.keys(res.body.user)).toContain('phone');
      });
  });

  test('Updates Portfolio/Project Details', async () => {
    expect.assertions(7);
    return request(app)
      .put(`/api/v1/users/update/projectInfo/${decadevCredentials.email}`)
      .set('Authorization', `Bearer ${decadevToken}`)
      .send({
        title: 'Rosetta Stone',
        languages: 'JavaScript, AngularJS, HTML, CSS, Bower, Git',
        id: projectId,
      })
      .expect(res => {
        expect(res.body.message).toEqual(expect.any(String));
        expect(Object.keys(res.body)).toContain('message');
        expect(Object.keys(res.body)).toContain('user');

        expect(Object.keys(res.body.user)).toContain('bio');
        expect(Object.keys(res.body.user)).toContain('github');
        expect(Object.keys(res.body.user)).toContain('email');
        expect(Object.keys(res.body.user)).toContain('phone');
      });
  });

  test('Updates Skill Details', async () => {
    expect.assertions(7);
    return request(app)
      .put(`/api/v1/users/update/skillInfo/${decadevCredentials.email}`)
      .set('Authorization', `Bearer ${decadevToken}`)
      .send({
        type: 'Stuff',
        description: 'CSS, HTML, Sass, TypeScript, JS',
        id: skillId,
      })
      .expect(res => {
        expect(res.body.message).toEqual(expect.any(String));
        expect(Object.keys(res.body)).toContain('message');
        expect(Object.keys(res.body)).toContain('user');

        expect(Object.keys(res.body.user)).toContain('bio');
        expect(Object.keys(res.body.user)).toContain('github');
        expect(Object.keys(res.body.user)).toContain('email');
        expect(Object.keys(res.body.user)).toContain('phone');
      });
  });

  test('Updates Employment Details', async () => {
    expect.assertions(7);
    return request(app)
      .put(`/api/v1/users/update/employmentInfo/${decadevCredentials.email}`)
      .set('Authorization', `Bearer ${decadevToken}`)
      .send({
        title: 'Back-End Engineer',
        duration: '2013 - PRESENT',
        location: 'Rosetta Stone',
        achievements: [
          'Built a language-learning app for Arabic learners in AngularJS',
          "Constructed a reporting tool so administrators could see their students' progress in real-time.",
          'Developed custom AngularJS services and corresponding tests.',
          'Created custom exact-target email templates and data extensions, and a tool to allow administrators to email users at the click of a button.',
        ],
        id: employmentId,
      })
      .expect(res => {
        expect(res.body.message).toEqual(expect.any(String));
        expect(Object.keys(res.body)).toContain('message');
        expect(Object.keys(res.body)).toContain('user');

        expect(Object.keys(res.body.user)).toContain('bio');
        expect(Object.keys(res.body.user)).toContain('github');
        expect(Object.keys(res.body.user)).toContain('email');
        expect(Object.keys(res.body.user)).toContain('phone');
      });
  });

  test('Deletes Employment details', async () => {
    expect.assertions(7);
    return request(app)
      .put(`/api/v1/users/update/delete-employment/${decadevCredentials.email}`)
      .set('Authorization', `Bearer ${decadevToken}`)
      .send({
        achievements: [
          'Built a language-learning app for Arabic learners in AngularJS',
          "Constructed a reporting tool so administrators could see their students' progress in real-time.",
          'Developed custom AngularJS services and corresponding tests.',
          'Created custom exact-target email templates and data extensions, and a tool to allow administrators to email users at the click of a button.',
        ],
        title: 'CTO',
        location: 'Google',
        duration: '2018 (9 months)',
        id: employmentId,
      })
      .expect(res => {
        expect(res.body.message).toEqual(expect.any(String));
        expect(Object.keys(res.body)).toContain('message');
        expect(Object.keys(res.body)).toContain('user');

        expect(Object.keys(res.body.user)).toContain('bio');
        expect(Object.keys(res.body.user)).toContain('github');
        expect(Object.keys(res.body.user)).toContain('email');
        expect(Object.keys(res.body.user)).toContain('phone');
      });
  });

  test('Deletes Skill Details', async () => {
    expect.assertions(7);
    return request(app)
      .put(`/api/v1/users/update/delete-skill/${decadevCredentials.email}`)
      .set('Authorization', `Bearer ${decadevToken}`)
      .send({
        type: 'Food',
        description: 'Cakes, Chocolate, IceCream',
        id: skillId,
      })
      .expect(res => {
        expect(res.body.message).toEqual(expect.any(String));
        expect(Object.keys(res.body)).toContain('message');
        expect(Object.keys(res.body)).toContain('user');

        expect(Object.keys(res.body.user)).toContain('bio');
        expect(Object.keys(res.body.user)).toContain('github');
        expect(Object.keys(res.body.user)).toContain('email');
        expect(Object.keys(res.body.user)).toContain('phone');
      });
  });

  test('Deletes Portfolio/Project Details', async () => {
    expect.assertions(7);
    return request(app)
      .put(`/api/v1/users/update/delete-project/${decadevCredentials.email}`)
      .set('Authorization', `Bearer ${decadevToken}`)
      .send({
        title: 'Rosetta Stone',
        languages: 'JavaScript, AngularJS, HTML, CSS, Bower, Grunt, Git',
        id: projectId,
      })
      .expect(res => {
        expect(res.body.message).toEqual(expect.any(String));
        expect(Object.keys(res.body)).toContain('message');
        expect(Object.keys(res.body)).toContain('user');

        expect(Object.keys(res.body.user)).toContain('bio');
        expect(Object.keys(res.body.user)).toContain('github');
        expect(Object.keys(res.body.user)).toContain('email');
        expect(Object.keys(res.body.user)).toContain('phone');
      });
  });
});
