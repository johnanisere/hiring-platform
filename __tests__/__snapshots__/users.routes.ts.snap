// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Updates Decadev Profile Adds new skill details of Decadevs 1`] = `
Object {
  "bio": "A very good dev",
  "count": 1,
  "education": Array [],
  "email": "decadev-user@example.com",
  "experience": Array [],
  "gender": "male",
  "github": "https://github.com",
  "hired": false,
  "interviews": Array [],
  "linkedIn": "https://linkedin.com/in/oluwaseyi-ogundijo-a29699195",
  "location": "Lagos",
  "name": "Jane Mary Lala",
  "phone": "08074583218",
  "pod": "java",
  "portfolio": Array [],
  "profilePhoto": "https://encq=tbn:ANd9GcSIcqjP8a8EfGrtkCX0CJXJ4HRltM5u8Jdhkz_TneWdb4rfPsMi",
  "publications": Array [],
  "role": "dev",
  "stackOverflow": "https://stackoverflow.com",
  "tests": Array [],
  "website": "https://www.google.com/",
}
`;
